core.coffee

```diff
@@ -440,6 +457,53 @@
 						url += '?' + queryOptions.join('&')
 					return url
 
+			compileBatch: (data) ->
+				if !utils.isArray(data)
+					throw new Error('Batch must be passed an array')
+				if !mimemessage
+					throw new Error('A mimemessage implementation supporting .factory must be supplied to perform batch operations')
+
+				buildReq = (req) =>
+					# This is a changeset
+					if utils.isArray(req)
+						csBody = req.map(buildReq)
+
+						return mimemessage.factory(
+							contentType: 'multipart/mixed'
+							body: csBody
+						)
+					# This is a normal request
+					contentTransferEncoding = 'binary'
+					if req.body?
+						contentType = 'application/json'
+						b = JSON.stringify(req.body)
+					else
+						contentType = 'application/http'
+						b = ''
+					splitApiRoot = @apiPrefix.split('/')
+					apiRoot = splitApiRoot[splitApiRoot.length - 2]
```

Comment:
What is this section about? What does it expect `@apiPrefix` to look like, and then for `apiRoot`/`target to look like?

Reply:
This was actually an over complicated way to try and replicate https://github.com/resin-io/pinejs-client-js/blob/master/core.coffee#L451-L452 with the caveat of not prepending the `apiPrefix` in case the url starts with a $n. This is for requests inside the changeset where the $n is substituted with the full expanded url obtained from the creation of another resource, so we don't need the prefix in that case. I rewrote the code, I think the intent is clearer now